src/deploy/functions/prepare.ts

```diff
@@ -37,6 +38,7 @@ export async function prepare(
   payload: args.Payload
 ): Promise<void> {
   const projectId = needProjectId(options);
+  const projectNumber = await getProjectNumber({ projectId });
```

Comment:
Check out definition for [`getProjectNumber`](https://github.com/firebase/firebase-tools/blob/8fc53ac15f0624aae86cca32c4f8eb438054b696/src/getProjectNumber.ts#L8).

It's designed to receive `options` here so that when you later call `getProjectNumber` with the same `options` object, you'd use the cached value instead of making yet another API call.

In fact, I'd actually suggest that we use [`needProjectNumber`](https://github.com/firebase/firebase-tools/blob/094faede699f7a9070ba641fdcbc10d1883803ba/src/projectUtils.ts#L84) instead.

Reply:
Ahh, thanks for pointing this out, fixed in the latest commit